init/redhat/waagent.service

```diff
@@ -5,6 +5,7 @@ After=network-online.target
 
 ConditionFileIsExecutable=/usr/sbin/waagent
 ConditionPathExists=/etc/waagent.conf
+ConditionVirtualization=microsoft
```

Comment:
This looks fine to me, although it does differ from the [patch we're carrying in Fedora](https://src.fedoraproject.org/rpms/WALinuxAgent/blob/rawhide/f/0001-waagent.service-set-ConditionVirtualization-microsof.patch) in that it's not a triggering conditional. 

@cjp256 do you still feel that it should be a triggering conditional? In my opinion, it's just as easy to `systemctl --edit --full waagent.service` as it is to drop in an additional triggering conditional, but I've not thought hard about it.